site/src/Map.jsx

```diff
@@ -1,135 +1,238 @@
+    const layersToShow = layers
+      .filter((layer) => visibleThemes.indexOf(layer.source) >= 0)
+      .map((layer) => layer.id);
+    setInteractiveLayerIds(layersToShow);
+  }, [visibleThemes]);
 
-  const onMouseEnter = useCallback(() => setCursor('pointer'), []);
-  const onMouseLeave = useCallback(() => setCursor('auto'), []);
+  const onMouseEnter = useCallback(() => setCursor("pointer"), []);
+  const onMouseLeave = useCallback(() => setCursor("auto"), []);
 
-  const onClick = useCallback(event => {
+  const onClick = useCallback((event) => {
     const feature = event.features && event.features[0];
-
     if (feature) {
-      setMapEntity(event.features[0].properties);
+      setMapEntity({
+        theme: feature.source,
+        type: feature.sourceLayer,
+        ...feature.properties,
+      });
     } else {
       setMapEntity({});
     }
   }, []);
 
-  function getMapStyle() {
-    let mapStyle;
-
-    if (pmTilesReady){
-     mapStyle =  mode === 'theme-light' ? MAP_STYLE : DARK_MAP_STYLE;
-    } else {
-      mapStyle = undefined;
-    }
-    return mapStyle;
-  }
-
   return (
     <>
       <div className={`map ${mode}`}>
         <MapLibreMap
-          id="myMap"
+          ref={mapRef}
           onMouseEnter={onMouseEnter}
           onMouseLeave={onMouseLeave}
+          onLoad={syncInteractiveLayerIds}
           onClick={onClick}
           cursor={cursor}
           hash={true}
+          mapStyle={MAP_STYLE}
           interactiveLayerIds={interactiveLayerIds}
           initialViewState={INITIAL_VIEW_STATE}
-          mapStyle={getMapStyle()}
-          style={{position: 'fixed', width: '100%', height: 'calc(100vh - 60px)'}}
+          style={{
+            position: "fixed",
+            width: "100%",
+            height: "calc(100vh - 60px)",
+          }}
+          attributionControl={false}
         >
-          <Source id="overture-places" type="vector" url={PLACES_PMTILES_URL}>
-            <Layer {...PLACES_MAP_STYLE} layout={{visibility: interactiveLayerIds.includes('places') ? 'visible' : 'none'}} />
-          </Source>
-          <NavigationControl position='top-right'></NavigationControl>
+          <ThemeSource name="buildings" url={PMTILES_URL} />
+          <ThemeSource name="places" url={PMTILES_URL} />
+          <ThemeSource name="divisions" url={PMTILES_URL} />
+
+          <Layer
+            id="divisions_division"
+            type="symbol"
+            source="divisions"
+            source-layer="division"
+            paint={{
+              "text-color": mode === "theme-light" ? "black" : "white",
+              "text-halo-color": mode === "theme-light" ? "white" : "black",
+              "text-halo-width": 1,
+            }}
+            layout={{
+              "text-font": ["Noto Sans Regular"],
+              "text-field": ["get", "@name"],
+              "text-size": 11,
+            }}
+          />
+
+          {visibleThemes.includes("divisions") ? (
```

Comment:
This was changed from using `layout.visibility` because that would override any of our defined `layout` properties, and using JSX lets react-map-gl perform the style diff. Any reason we would prefer changing `visibility` instead of removing the layer?

Reply:
The only consideration I can think of is performance. Perhaps the overhead of adding/remove the layers is greater than keeping them loaded but invisible?

Once we get the land cover layers loaded in we can re-examine whether any performance optimizations are required from a rendering perspective.

Reply:
The base/land_cover layers are in now the PR respecting the `cartography.max_zoom / min_zoom` attributes. 

Removing/adding the layer does make HTTP requests but those will be cached by all browsers but Firefox. The same is true of modifying `visibility` at the Style level.